33-lifecycle-lab_v16/src/03-custominput.jsx

```diff
@@ -1,10 +1,80 @@
-/**
- * Implment an input component that shows 4
- * input boxes writing user input in the currently
- * focused panel
- *
- * See lab description on the webpage for a live example
- * how this should work.
- */
-
-
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default class SquareFocus extends React.Component{
+
+    constructor(props) {
+        super(props);
+        this.state = {texts : []};
+    }
+
+
+    setFocus = (el,index) =>{
+        if(el!=null){
+            el.focus()
+            el.style.backgroundColor = "#ccc";
+        }
+    }
+
+    fillText = (e,index) =>{
+        let isFocused = (document.activeElement === e.target);
+        if(isFocused){
+            if(document.activeElement.innerHTML != ''){
+                let updatedArr = this.state.texts;
+                updatedArr[index] = {txt : String.fromCharCode(e.keyCode)};
+                this.setState({texts:updatedArr});
+            }else{
+            this.setState({texts:[...this.state.texts , {txt : String.fromCharCode(e.keyCode)} ]});
+            }
+        }
+    }
+
+    componentDidUpdate(){
+          let activeEl = document.activeElement;
+          activeEl.style.backgroundColor = "white";
+          activeEl.blur();
+          let parentDiv =  this.parent
+          if(activeEl.nextSibling!=null){
+           this.setFocus(activeEl.nextSibling);

```

Comment:
Notice `setFocus` will change the `state`, which will lead back to `componentDidUpdate`. This circle is an anti pattern. The hook `componentDidUpdate` should ONLY be used to make DOM modifications in response to state or props change that were not possible in render.
(For example you could use it to call `focus` on a DOM element)